source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -214,12 +214,17 @@ void CheckRequestUtils::setAttrContextRequest(
                  max_request_bytes, pack_as_bytes, encode_raw_headers, request_header_matchers);
 }
 
-void CheckRequestUtils::setTLSSession(
-    envoy::service::auth::v3::AttributeContext::TLSSession& session,
-    const Ssl::ConnectionInfoConstSharedPtr ssl_info) {
-  if (!ssl_info->sni().empty()) {
-    const std::string server_name(ssl_info->sni());
-    session.set_sni(server_name);
+void setTLSSession(envoy::service::auth::v3::AttributeContext::TLSSession& session,
+                   OptRef<const Envoy::Network::Connection> connection) {
+  if (connection.has_value()) {
+    const Ssl::ConnectionInfoConstSharedPtr ssl_info = connection->ssl();
+    if (ssl_info != nullptr && !ssl_info->sni().empty()) {
+      const std::string server_name(ssl_info->sni());
+      session.set_sni(server_name);
+    } else if (!connection->requestedServerName().empty()) {
+      const std::string server_name(connection->requestedServerName());
+      session.set_sni(server_name);
+    }
```

Comment:
```suggestion
void setTLSSession(envoy::service::auth::v3::AttributeContext::TLSSession& session,
                   const Network::Connection& connection) {
    const Ssl::ConnectionInfoConstSharedPtr ssl_info = connection.ssl();
    if (ssl_info != nullptr && !ssl_info->sni().empty()) {
      const std::string server_name(ssl_info->sni());
      session.set_sni(server_name);
    } else if (!connection.requestedServerName().empty()) {
      const std::string server_name(connection.requestedServerName());
      session.set_sni(server_name);
```

(This isn't correctly formatted; don't accept it directly. Take it and run clang-format.)

Reply:
https://github.com/envoyproxy/envoy/pull/34100/commits/a49951c222c53f3ef2cde3a1b9843582ca9b39d8